src/containers/tokens/components/bonding-curve-form/index.js

```diff
@@ -0,0 +1,422 @@
+import { connect } from 'react-redux'
+import { toBN } from 'ethjs'
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { formValueSelector } from 'redux-form'
+
+import Button from '../../../../components/button'
+import { form } from '../../../../utils/form-generator'
+import { required, number, positiveNumber } from '../../../../utils/validation'
+import {
+  decimalStringToWeiBN,
+  weiBNToDecimalString
+} from '../../../../utils/number'
+import * as bondingCurveSelectors from '../../../../reducers/bonding-curve'
+import * as bondingCurveActions from '../../../../actions/bonding-curve'
+
+const {
+  Form: BuyPNKFromBondingCurveForm,
+  isInvalid: getBuyPNKFromBondingCurveFormIsInvalid,
+  submit: submitBuyPNKFromBondingCurveForm
+} = form('buyPNKFromBondingCurveForm', {
+  header: {
+    type: 'header',
+    props: { title: 'BUY PNK' }
+  },
+  amountOfETH: {
+    type: 'text',
+    validate: [required, number, positiveNumber],
+    props: {
+      type: 'number',
+      className: 'Form-noMargins'
+    }
+  },
+  rate: {
+    type: 'info',
+    props: {
+      className: 'Tokens-form-rate'
+    }
+  }
+})
+
+const { Form: SellPNKToBondingCurveFormHeader } = form(
+  'sellPNKToBondingCurveFormHeader',
+  {
+    header: {
+      type: 'header',
+      props: { title: 'SELL PNK' }
+    }
+  }
+)
+
+const {
+  Form: SellPNKToBondingCurveForm,
+  isInvalid: getSellPNKToBondingCurveFormIsInvalid,
+  submit: submitSellPNKToBondingCurveForm
+} = form('sellPNKToBondingCurveForm', {
+  amountOfPNK: {
+    type: 'text',
+    validate: [required, number, positiveNumber],
+    props: {
+      type: 'number',
+      className: 'Form-noMargins'
+    }
+  },
+  rate: {
+    type: 'info',
+    props: {
+      className: 'Tokens-form-rate'
+    }
+  }
+})
+
+/**
+ * Convert a fraction to float.
+ * @param {BigNumber} numerator - The numerator.
+ * @param {BigNumber} denominator - The denominator.
+ * @returns {number} - The result.
+ */
+function toFloat(numerator, denominator) {
+  return (
+    numerator
+      .mul(toBN(10000))
+      .div(denominator)
+      .toNumber() / 10000
+  )
+}
+
+/**
+ * Round number to 4 significant figures.
+ * @param {string} n - Input number.
+ * @returns {string} - Rounded number with at least 4 significant figures.
+ */
+function round(n) {
```

Comment:
Move this to a `util`. Also nothing is being rounded, it is just truncated after 4 decimal places. Rename

Reply:
Will change as suggested.